src/jvm/test/org/jetbrains/kotlinx/lincheck_test/representation/RunWithLambdaRepresentationTests.kt

```diff
@@ -0,0 +1,309 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck_test.representation
+
+import org.jetbrains.kotlinx.lincheck.LincheckAssertionError
+import org.jetbrains.kotlinx.lincheck.runConcurrentTest
+import org.jetbrains.kotlinx.lincheck.util.UnsafeHolder
+import org.jetbrains.kotlinx.lincheck_test.gpmc.SpinLock
+import org.jetbrains.kotlinx.lincheck_test.gpmc.withLock
+import org.jetbrains.kotlinx.lincheck_test.util.checkLincheckOutput
+import org.jetbrains.kotlinx.lincheck_test.util.isJdk8
+import org.junit.Test
+import java.util.concurrent.ConcurrentLinkedDeque
+import java.util.concurrent.Executors
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.atomic.AtomicIntegerArray
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater
+import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.atomic.AtomicLongArray
+import java.util.concurrent.atomic.AtomicReference
+import java.util.concurrent.atomic.AtomicReferenceArray
+import kotlin.concurrent.thread
+import kotlin.random.Random
+
+
+abstract class BaseRunWithLambdaRepresentationTest<R>(private val outputFileName: String) {
```

Comment:
Please replace `RunWithLabmda` to something more consistent with the API (please remember about the corresponding file names)

Reply:
Renamed to `RunConcurrentRepresentationTest`. Also, moved the corresponding trace `.txt` files into dedicated sub-folder.